cdf_fabric_replicator/extractor.py

```diff
@@ -25,6 +30,40 @@ def __init__(self, stop_event: CancellationToken) -> None:
         self.azure_credential = DefaultAzureCredential()
         self.stop_event = stop_event
 
+    def run(self) -> None:
+        self.config = self.get_current_config()
+        self.client = self.config.cognite.get_cognite_client("cdf-fabric-extractor")
+        self.state_store = self.get_current_statestore()
+        self.state_store.initialize()
+
+        if not self.config.source:
+            self.logger.error("No source path or directory provided")
+            return
+
+        while self.stop_event.is_set() is False:
+            token = self.azure_credential.get_token("https://storage.azure.com/.default").token
+
+            self.run_extraction_pipeline(status="seen")
+
+            if self.config.source.raw_time_series_path and self.config.destination.time_series_prefix:
+                time_series_data = self.convert_lakehouse_data_to_df(
+                    self.config.source.abfss_prefix + "/" + self.config.source.raw_time_series_path, token=token
+                )
+                self.write_time_series_to_cdf(time_series_data)
+
+            if self.config.source.event_path:
+                state_id = f"{self.config.source.event_path}-state"
+                self.write_event_data_to_cdf(
+                    self.config.source.abfss_prefix + "/" + self.config.source.event_path,
+                    token=token,
+                    state_id=state_id,
+                )
+
+            if self.config.source.file_path:
+                self.upload_files_from_abfss(self.config.source.abfss_prefix + "/" + self.config.source.file_path)
+
+            time.sleep(5)
```

Comment:
Does not have to be this pr

The time.sleep(5) at the end of the method is likely used to prevent the method from running too frequently, effectively acting as a rate limiter. However, using time.sleep is a simple solution and may not be the most efficient or reliable way to handle rate limiting.

A more robust solution would be to implement a proper rate limiting mechanism. This could be done using a library like ratelimiter in Python.

Reply:
Are we able to add that 3rd party library to the Cognite repo?: